cpp/arcticdb/version/local_versioned_engine.cpp

```diff
@@ -1075,9 +1098,61 @@ std::vector<AtomKey> LocalVersionedEngine::batch_write_internal(
             validate_index
         ));
     }
-    return folly::collect(results_fut).get();
+    return results_fut;
+}
+
+std::pair<VersionId, std::shared_ptr<DeDupMap>> LocalVersionedEngine::create_version_id_and_dedup_map(
+    const version_store::UpdateInfo& update_info, 
+    const StreamId& stream_id, 
+    const WriteOptions& write_options){
+    if(cfg().write_options().de_duplication()) {
+        return {update_info.next_version_id_, get_de_dup_map(stream_id, update_info.previous_index_key_, write_options)};
+    } else {
+        return {update_info.next_version_id_, std::make_shared<DeDupMap>()};
+    }
 }
 
+std::vector<VersionedItem> LocalVersionedEngine::batch_write_versioned_dataframe_internal(
+    const std::vector<StreamId>& stream_ids,
+    std::vector<InputTensorFrame>&& frames,
+    bool prune_previous_versions,
+    bool validate_index
+) {
+    auto write_options = get_write_options();
+    auto update_info_fut_vector = async_batch_get_latest_undeleted_version_and_next_version_id(store(),
+                                                                                               version_map(),
+                                                                                               stream_ids);
+    util::check(stream_ids.size() == update_info_fut_vector.size(), "stream_ids and update_info_fut_vectormust be of the same size");                                                                                    
```

Comment:
Please use the new error code framework rather than `util::check`

Reply:
no, I would rather remove this assertion. it is auto testing logic, no user should ever encounter it

Reply:
Checking supposed invariants that users should never encounter is something we frequently do in the codebase. Otherwise, when there is a bug that would have failed the check, you end up with much harder to debug errors like segfaults.